wmi_check/datadog_checks/wmi_check/wmi_check.py

```diff
@@ -14,68 +15,49 @@ class WMICheck(WinWMICheck):
     Windows only.
     """
 
-    def __init__(self, name, init_config, agentConfig, instances):
-        WinWMICheck.__init__(self, name, init_config, agentConfig, instances)
-        self.wmi_samplers = {}
-        self.wmi_props = {}
-
-    def check(self, instance):
+    def __init__(self, name, init_config, instances):
+        # type: (str, Dict[str, Any], List[Dict[str, Any]]) -> None
+        super(WMICheck, self).__init__(name, init_config, instances)
+        self.custom_tags = self.instance.get('tags', [])  # type: List[str]
+        self.filters = self.instance.get('filters', [])  # type: List[Dict[str, WMIFilter]]
+        self.metrics_to_capture = self.instance.get('metrics', [])  # type: List[List[str]]
+        self.tag_by = self.instance.get('tag_by', "")  # type: str
+        self.tag_queries = self.instance.get('tag_queries', [])  # type: List[TagQuery]
+
+    def check(self, _):
+        # type: (Any) -> None
         """
         Fetch WMI metrics.
         """
-
-        # Connection information
-        host = instance.get('host', "localhost")
-        namespace = instance.get('namespace', "root\\cimv2")
-        provider = instance.get('provider')
-        username = instance.get('username', "")
-        password = instance.get('password', "")
-
-        # WMI instance
-        wmi_class = instance.get('class')
-        metrics = instance.get('metrics')
-        filters = instance.get('filters')
-        tag_by = instance.get('tag_by', "")
-        tag_queries = instance.get('tag_queries', [])
-
-        constant_tags = instance.get('constant_tags')
-        custom_tags = instance.get('tags', [])
-        if constant_tags is None:
-            constant_tags = list(custom_tags)
-        else:
-            constant_tags.extend(custom_tags)
+        constant_tags = self.instance.get('constant_tags', [])
+        if constant_tags:
             self.log.warning("`constant_tags` is being deprecated, please use `tags`")
+        constant_tags.extend(self.custom_tags)
```

Comment:
Any reason not moving `constant_tags` to `__init__` too ?

Reply:
Yes, apparently warnings do not appear in the UI if they're on `__init__` instead of `check`

Reply:
But seems to be a regular log here `log.warning`, not `warnings.warn` (https://docs.python.org/3/library/warnings.html).

Reply:
ohhh, then I'll change it